store.go

```diff
@@ -2842,57 +2846,69 @@ func (s *store) mount(id string, options drivers.MountOpts) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	if options.UidMaps != nil || options.GidMaps != nil {
-		options.DisableShifting = !s.canUseShifting(options.UidMaps, options.GidMaps)
-	}
 
-	// function used to have a scope for rlstore.StopWriting()
-	tryMount := func() (string, error) {
-		if err := rlstore.startWriting(); err != nil {
-			return "", err
-		}
-		defer rlstore.stopWriting()
-		if rlstore.Exists(id) {
-			return rlstore.Mount(id, options)
-		}
-		return "", nil
+	if err := validateMountOptions(mountOpts); err != nil {
+		return "", err
 	}
-	mountPoint, err := tryMount()
-	if mountPoint != "" || err != nil {
-		return mountPoint, err
+	options := drivers.MountOpts{
+		MountLabel: mountLabel,
+		Options:    append(mountOpts, "ro"),
 	}
+	var imageHomeStore roImageStore // Set if image != ""
+	var cimage *Image               // Set if image != ""
 
-	// check if the layer is in a read-only store, and return a better error message
-	for _, store := range lstores {
+	if err := rlstore.startWriting(); err != nil {
+		return "", err
+	}
+	defer rlstore.stopWriting()
+	for _, s := range lstores {
+		store := s
```

Comment:
(Loop iteration variables are now fresh in each copy, so this should be no longer required anywhere.)

Reply:
(Also `for _, s := range s.roImageStores `)